__DEFINES/setup.dm

```diff
@@ -1112,6 +1112,7 @@ var/default_colour_matrix = list(1,0,0,0,\
 #define PURCHASER		256 //it connects to the centcom database at roundstart
 #define WIREJACK		512 //can we wirejack it? if flagged, machine calls wirejack()
 #define SHUTTLEWRENCH	1024 //if this flag exists, the computer can be wrenched on shuttle floors
+#define SECUREDPANEL 2048 //it won't let you open the deconstruction panel if you don't have the linked account number. Originally used for custom vending machines
```

Comment:
Honestly this seems like an extremely specific flag.

What if you made it so that this is for general `allowed()` and made the custom vendors override `allowed()`?